### PR TITLE
Allow agent.jar to be run in docker image

### PR DESCRIPTION
**Pull request by kvish**
Jira Issues: DCOS_OSS-5064

### CODE REVIEW

**Comment by kvish** [comment]
blocked on https://github.com/mesosphere/usi/pull/98

File: src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java [hunk 1]
```diff
@@ -78,6 +81,7 @@ public MesosAgentSpecTemplate(
     this.jvmArgs = jvmArgs;
     this.additionalURIs = additionalURIs;
     this.nodeProperties = nodeProperties;
+    this.containerImage = StringUtils.isNotBlank(containerImage) ? containerImage : null;
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java]: Why not use `Optional.OfNullable` here instead?

File: src/test/java/org/jenkinsci/plugins/mesos/integration/MesosApiTest.java [hunk 2]
```diff
@@ -41,11 +43,19 @@
   static ActorSystem system = ActorSystem.create("mesos-scheduler-test");
   static ActorMaterializer materializer = ActorMaterializer.create(system);
 
+  static MesosAgentConfig config =
+      new MesosAgentConfig(
+          "linux",
+          "mesos",
+          Option.apply("filesystem/linux,docker/runtime"),
```
Comment (jeschkies) [inline src/test/java/org/jenkinsci/plugins/mesos/integration/MesosApiTest.java]: Is this working on macOS?

Comment (kvish) [reply src/test/java/org/jenkinsci/plugins/mesos/integration/MesosApiTest.java]: parameterized to only run docker test in linux

File: src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java (hunk 1 above)
Comment (kvish) [reply src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java]: changed to move optional here

File: src/test/java/org/jenkinsci/plugins/mesos/integration/DockerAgentTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  @RegisterExtension
+  static MesosClusterExtension mesosCluster =
+      MesosClusterExtension.builder()
+          .withMesosMasterUrl(String.format("zk://%s/mesos", zkServer.getConnectionUrl()))
+          .withLogPrefix(MesosApiTest.class.getCanonicalName())
+          .withAgentConfig(config)
+          .build(system, materializer);
+
+  @Test
+  @EnabledOnOs(OS.LINUX)
```
Comment (jeschkies) [inline src/test/java/org/jenkinsci/plugins/mesos/integration/DockerAgentTest.java:49]: Nice!

**Review by jeschkies — APPROVED** [review]
Thanks a ton!

**Review by takirala — APPROVED** [review]

**Pull request merged**